### PR TITLE
SONARJS-830 Rule: Return values should not be ignored when function calls don't have any side effects

### PR DESCRIPTION
**Pull request by ivandalbosco**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/UselessStringOperationCheck.java [hunk 1]
```diff
@@ -36,6 +36,7 @@
 import org.sonar.plugins.javascript.api.visitors.SubscriptionVisitorCheck;
 
 @Rule(key = "S1154")
+@Deprecated  // by S2201
```
Comment (SonarTech) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/UselessStringOperationCheck.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: javascript-checks/src/main/java/org/sonar/javascript/checks/UselessStringOperationCheck.java [hunk 2]
```diff
@@ -36,6 +36,7 @@
...
 public class UselessStringOperationCheck extends SubscriptionVisitorCheck {
```
Comment (SonarTech) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/UselessStringOperationCheck.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 1 major
* MINOR 2 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [CheckList.java#L188](https://github.com/SonarSource/sonar-javascript/blob/fa9736811c6754863f456d1fd8708e6a4e2466a6/javascript-checks/src/main/java/org/sonar/javascript/checks/CheckList.java#L188): Remove this use of "UselessStringOperationCheck"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: javascript-checks/src/main/java/org/sonar/javascript/checks/UselessStringOperationCheck.java [hunk 3]
```diff
@@ -35,7 +35,11 @@
 import org.sonar.plugins.javascript.api.tree.statement.ExpressionStatementTree;
 import org.sonar.plugins.javascript.api.visitors.SubscriptionVisitorCheck;
 
+/**
+ * @deprecated by S2201
+ */
 @Rule(key = "S1154")
+@Deprecated
 public class UselessStringOperationCheck extends SubscriptionVisitorCheck {
```
Comment (SonarTech) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/UselessStringOperationCheck.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [CheckList.java#L188](https://github.com/SonarSource/sonar-javascript/blob/1580de9b82751ff40c54b625a1d807ece92b4bd7/javascript-checks/src/main/java/org/sonar/javascript/checks/CheckList.java#L188): Remove this use of "UselessStringOperationCheck"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: its/ruling/src/test/expected/javascript-S2201.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+'project:dojo-1.8.1/dojox/dtl.js.uncompressed.js':[
+93,
```
Comment (ivandalbosco) [inline its/ruling/src/test/expected/javascript-S2201.json]: @pynicolas Good catch! Thanks.

File: its/ruling/src/test/expected/javascript-S2201.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+146,
+],
+'project:ecmascript6/snoode/src/endpoints/v1.es6.js':[
+585,
+],
+'project:yui-3.7.3/charts-base/charts-base.js':[
+4618,
+],
+'project:yui-3.7.3/imageloader/imageloader.js':[
+383,
```
Comment (ivandalbosco) [inline its/ruling/src/test/expected/javascript-S2201.json]: @pynicolas OK

**Pull request merged**